scripts/artifactory-cleanup.js

```diff
@@ -0,0 +1,69 @@
+const exec = require('child_process').exec;
+const semver = require('semver');
+const packageJson = require('../package.json');
+
+const BASE_ARTIFATORY_URL = 'https://artifactory.wikia-inc.com/artifactory';
+const BASE_PACKAGE_PATH = '/npm-local/@fandom/jwplayer-fandom/-/@fandom';
+const BASE_READ_PATH = `/api/storage${BASE_PACKAGE_PATH}`;
+const BASE_DELETE_PATH = BASE_PACKAGE_PATH;
+
+const currentMajorVersion = semver.major(packageJson.version);
+const previousMinorVersion = semver.minor(packageJson.version) - 1;
+const versionThreshold = `${currentMajorVersion}.${previousMinorVersion}.0`;
+
+async function getArtifactoryTokenFromYarnConfig() {
+	return new Promise((resolve, reject) => {
```

Comment:
When you use async in front of a function, it automatically wraps the returned value in a Promise. So if you explicitly return a new Promise inside an async function, you'd be wrapping a Promise inside another Promise, which is usually unnecessary and can complicate your code.